staging/src/k8s.io/apimachinery/pkg/util/httpstream/spdy/connection.go

```diff
@@ -34,38 +35,50 @@ type connection struct {
 	streams          []httpstream.Stream
 	streamLock       sync.Mutex
 	newStreamHandler httpstream.NewStreamHandler
+	pingsSent        int64
 }
 
 // NewClientConnection creates a new SPDY client connection.
-func NewClientConnection(conn net.Conn) (httpstream.Connection, error) {
+//
+// If pingPeriod is non-zero, a background goroutine will send periodic Ping
+// frames to the server. Use this to keep idle connections through certain load
+// balancers alive longer.
+func NewClientConnection(conn net.Conn, pingPeriod time.Duration) (httpstream.Connection, error) {
```

Comment:
Please instead add sibling methods, then you don't have to modify call sites (or break downstream users, if there are any). (E.g., NewClientConnectionWithPings())

Reply:
Great idea, done